### PR TITLE
[Benchmarks] support CPU counter measurements

### PR DESCRIPTION
**Pull request by igchor**
for api_overhead_benchmark_ur

### CODE REVIEW

File: scripts/benchmarks/benches/compute.py [hunk 1]
```diff
@@ -71,17 +71,20 @@ def benchmarks(self) -> list[Benchmark]:
...
+                SubmitKernelURWithTime(self, 1),
+                SubmitKernelURWithCPUCounter(self, 0),
+                SubmitKernelURWithCPUCounter(self, 1),
             ]
 
         return benches
 
 class ComputeBenchmark(Benchmark):
-    def __init__(self, bench, name, test):
+    def __init__(self, bench, name, test, unit="time [us]"):
```
Comment (pbalcer) [inline scripts/benchmarks/benches/compute.py:83]: this looks very odd, please do an "enum" ([example](https://github.com/oneapi-src/unified-runtime/blob/main/scripts/benchmarks/benches/options.py#L4))

File: scripts/benchmarks/benches/compute.py [hunk 2]
```diff
@@ -175,7 +210,7 @@ def __init__(self, bench, ioq, isCopyOnly, source, destination, size):
         self.source = source
         self.destination = destination
         self.size = size
-        super().__init__(bench, "api_overhead_benchmark_sycl", "ExecImmediateCopyQueue")
+        super().__init__(bench, "api_overhead_benchmark_sycl", "ExecImmediateCopyQueue", "time [us]")
```
Comment (pbalcer) [inline scripts/benchmarks/benches/compute.py:213]: You've set a default unit, do you need to set these?

File: scripts/benchmarks/benches/compute.py [hunk 3]
```diff
@@ -148,14 +162,35 @@ def bin_args(self) -> list[str]:
             "--KernelExecTime=1"
         ]
 
-class SubmitKernelUR(ComputeBenchmark):
+class SubmitKernelURWithTime(ComputeBenchmark):
```
Comment (pbalcer) [inline scripts/benchmarks/benches/compute.py:165]: I don't think you need to duplicate these. Simply return two values from the run() with the extra label. In fact, I think all you have to do is remove the if at line 117 and everything will just work without duplicated benchmark classes.

File: scripts/benchmarks/benches/compute.py (hunk 2 above)
Comment (igchor) [reply scripts/benchmarks/benches/compute.py]: Not longer needed.

File: scripts/benchmarks/benches/compute.py (hunk 3 above)
Comment (igchor) [reply scripts/benchmarks/benches/compute.py]: I also needed to remove unit() function and set it on the result directly. Done.

File: scripts/benchmarks/benches/compute.py [hunk 4]
```diff
@@ -77,6 +77,13 @@ def benchmarks(self) -> list[Benchmark]:
 
         return benches
 
+def parse_unit_type(compute_unit):
+    if "[count]" in compute_unit:
+        return "CPU count"
```
[the remark this answers is not in the document]
Comment (igchor) [reply scripts/benchmarks/benches/compute.py]: Sure, done.

**Review by pbalcer — APPROVED** [review]

**Pull request merged**